Source/Urho3D/Container/Vector.h

```diff
@@ -852,7 +852,7 @@ template <class T> class PODVector : public VectorBase
     {
         unsigned oldSize = size_;
         Resize(size_ + vector.size_);
-        CopyElements(Buffer() + oldSize, vector.Buffer(), vector.size_);
+        CopyElements(Buffer() + oldSize, vector.Buffer(), (this == &vector) ? oldSize : vector.size_);

```

Comment:
How about:
```
// Obtain the size before resizing, in case the other vector is another reference to this vector
unsigned thisSize = size_;
unsigned vectorSize = vector.size_;
Resize(thisSize + vectorSize);
CopyElements(Buffer() + thisSize, vector.Buffer(), vectorSize);
```

Reply:
++, this one looks cleaner.

Reply:
Changed it to that version, I agree, it's cleaner.